frontend/src/app/Settings/Facility/Components/OrderingProvider.tsx

```diff
@@ -2,39 +2,33 @@ import React from "react";
 
 import { stateCodes } from "../../../../config/constants";
 import { requiresOrderProvider } from "../../../utils/state";
-import { ValidateField } from "../FacilityForm";
-import { FacilityErrors } from "../facilitySchema";
 import Dropdown from "../../../commonComponents/Dropdown";
 import TextInput from "../../../commonComponents/TextInput";
-import { getSubStrAfterChar } from "../../../utils/text";
+import { phoneNumberIsValid } from "../../../patients/personSchema";
 
 interface Props {
-  facility: Facility;
-  updateProvider: (provider: Provider) => void;
-  errors: FacilityErrors;
-  validateField: ValidateField;
   newOrg?: boolean;
+  errors: any;
+  register: any;
+  formCurrentValues: any;
 }
 
 const OrderingProvider: React.FC<Props> = ({
-  facility,
-  updateProvider,
-  errors,
-  validateField,
   newOrg = false,
+  errors,
+  register,
+  formCurrentValues,
 }) => {
-  const onChange = (
-    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
-  ) => {
-    let fieldName = getSubStrAfterChar(e.target.name, "-");
-    updateProvider({ ...provider, [fieldName]: e.target.value });
-  };
+  const isRequired = requiresOrderProvider(
+    formCurrentValues.facility.state || ""
+  );
 
-  const { orderingProvider: provider } = facility;
-  const isRequired = requiresOrderProvider(facility.state || "");
+  const isValidNPI = (npi: string) => {
+    return /^\d{10}$/.test(npi);
```

Comment:
⚠️ previously the regex to validate the NPI was [`/^\d{1,10}$/`](https://github.com/CDCgov/prime-simplereport/blob/97b716fcb0c73f982b9b35bc85621b536f73f7a1/frontend/src/app/Settings/Facility/facilitySchema.ts#LL35C6-L35C6), which means the NPI could be set to anything with less than 10 digits which according to our [error message](https://github.com/CDCgov/prime-simplereport/blob/97b716fcb0c73f982b9b35bc85621b536f73f7a1/frontend/src/app/Settings/Facility/facilitySchema.ts#LL45C11-L45C11) is incorrect.

Let me know if I should change this back to the previous regex.